tests/integration/controller/manifest/manifest_controller_test.go

```diff
@@ -6,68 +6,112 @@ import (
+	It("setup OCI", func() {
+		err := testutils.PushToRemoteOCIRegistry(server, manifestFilePath, installName)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	BeforeEach(
+		func() {
+			Expect(os.RemoveAll(filepath.Join(os.TempDir(), ociTempDir))).To(Succeed())
+		},
+	)
+}
+
 var _ = Describe(
-	"Rendering manifest install layer", func() {
-		mainOciTempDir := "main-dir"
-		installName := filepath.Join(mainOciTempDir, "installs")
-		It(
-			"setup OCI", func() {
-				manifesttest.PushToRemoteOCIRegistry(installName)
-			},
-		)
-		BeforeEach(
-			func() {
-				Expect(os.RemoveAll(filepath.Join(os.TempDir(), mainOciTempDir))).To(Succeed())
-			},
-		)
-		DescribeTable(
-			"Test OCI specs",
-			func(
-				givenCondition func(manifest *v1beta2.Manifest) error,
-				expectFunctions ...func(manifestName string) error,
-			) {
-				manifest := testutils.NewTestManifest("oci")
-				Eventually(givenCondition, standardTimeout, standardInterval).
-					WithArguments(manifest).Should(Succeed())
+	"Rendering manifest install layer", Ordered, func() {
+		ociTempDir := "main-dir"
+		installName := filepath.Join(ociTempDir, "installs")
 
-				for _, expectFn := range expectFunctions {
-					Eventually(expectFn, standardTimeout, standardInterval).
-						WithArguments(manifest.GetName()).Should(Succeed())
-				}
+		setupTestEnvironment(ociTempDir, installName)
 
-				Eventually(manifesttest.DeleteManifestAndVerify(manifest), standardTimeout, standardInterval).Should(Succeed())
-			},
-			Entry(
-				"When Manifest CR contains a valid install OCI image specification, "+
-					"expect state in ready",
-				manifesttest.WithValidInstallImageSpec(installName, false, false),
-				manifesttest.ExpectManifestStateIn(shared.StateReady),
-				manifesttest.ExpectOCISyncRefAnnotationExists(true),
-			),
-			Entry(
-				"When Manifest CR contains a valid install OCI image specification and enabled deploy resource, "+
-					"expect state in ready",
-				manifesttest.WithValidInstallImageSpec(installName, true, false),
-				manifesttest.ExpectManifestStateIn(shared.StateReady),
-				manifesttest.ExpectOCISyncRefAnnotationExists(true),
-			),
-			Entry(
-				"When Manifest CR contains an invalid install OCI image specification, "+
-					"expect state in error",
-				manifesttest.WithInvalidInstallImageSpec(false),
-				manifesttest.ExpectManifestStateIn(shared.StateError),
-				manifesttest.ExpectOCISyncRefAnnotationExists(false),
-			),
-		)
+		Context("Given a Manifest CR", func() {
+			It("When Manifest CR contains a valid install OCI image specification",
+				func() {
+					manifest := testutils.NewTestManifest("oci")
+
+					Eventually(testutils.WithValidInstallImageSpec(ctx, controlPlaneClient, installName,
+						manifestFilePath,
+						serverAddress, false, false), standardTimeout, standardInterval).
+						WithArguments(manifest).
+						Should(Succeed())
+					By("Then Manifest CR is in Ready State", func() {
```

Comment:
Consider moving this "When" into a dedicated `It(` container, so that it is on the same nesting level as the "When".
It may require to replace the `BeforeEach` with `BeforeAll`